src/main/webapp/app/code-editor/service/code-editor-repository.service.ts

```diff
@@ -37,21 +40,43 @@ export interface IBuildLogService {
 
 @Injectable({ providedIn: 'root' })
 export class CodeEditorRepositoryService extends DomainDependentEndpoint implements ICodeEditorRepositoryService {
-    constructor(http: HttpClient, jhiWebsocketService: JhiWebsocketService, domainService: DomainService) {
+    constructor(http: HttpClient, jhiWebsocketService: JhiWebsocketService, domainService: DomainService, private conflictService: CodeEditorConflictStateService) {
         super(http, jhiWebsocketService, domainService);
     }
 
-    isClean = () => {
-        return this.http.get<any>(this.restResourceUrl!).map(data => ({ isClean: data.isClean }));
+    getStatus = () => {
+        return this.http.get<any>(this.restResourceUrl!).pipe(
+            this.handleErrorResponse<{ repositoryStatus: string }>(),
+            tap(({ repositoryStatus }) => {
+                if (repositoryStatus !== CommitState.CONFLICT) {
+                    this.conflictService.notifyConflictState(GitConflictState.OK);
+                }
+            }),
+        );
     };
 
     commit = () => {
-        return this.http.post<void>(`${this.restResourceUrl}/commit`, {});
+        return this.http.post<void>(`${this.restResourceUrl}/commit`, {}).pipe(this.handleErrorResponse());
     };
 
     pull = () => {
-        return this.http.get<void>(`${this.restResourceUrl}/pull`, {});
+        return this.http.get<void>(`${this.restResourceUrl}/pull`, {}).pipe(this.handleErrorResponse());
+    };
+
+    // TODO: Maybe we don't have to check for the error here?
```

Comment:
Can you fix the TODO before we merge?

Reply:
This was already done, I added a comment to this method.